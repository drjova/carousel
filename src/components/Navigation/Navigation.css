#navigation {
  margin: 0 auto;
}

#navigation button {
  padding: 10px;
  font-weight: bold;
  font-size: 15px;
  margin: 5px;
  background-color: #4a739e;
  color: #fff;
}
