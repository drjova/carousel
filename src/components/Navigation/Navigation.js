import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Navigation.css';

class Navigation extends Component {
  renderPrevArrow() {
    const { total, current, onClickPrev } = this.props;

    if (current > 0) {
      return <button onClick={onClickPrev}>Prev</button>;
    }

    return null;
  }
  renderNextArrow() {
    const { total, current, onClickNext } = this.props;
    if (current < total - 1) {
      return <button onClick={onClickNext}>Next</button>;
    }
    return null;
  }

  render() {
    return (
      <div id="navigation">
        {this.renderPrevArrow()}
        {this.renderNextArrow()}
      </div>
    );
  }
}

Navigation.propTypes = {
  total: PropTypes.number.isRequired,
  current: PropTypes.number.isRequired,
  onClickNext: PropTypes.func.isRequired,
  onClickPrev: PropTypes.func.isRequired,
};

export default Navigation;
