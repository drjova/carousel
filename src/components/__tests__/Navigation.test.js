import React from 'react';
import { mount } from 'enzyme';

import Navigation from '../Navigation';

describe('Navigation component', () => {
  it('should render with both buttons', () => {
    const component = mount(
      <Navigation
        current={2}
        total={12}
        onClickPrev={jest.fn()}
        onClickNext={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render only previous button', () => {
    const component = mount(
      <Navigation
        current={11}
        total={12}
        onClickPrev={jest.fn()}
        onClickNext={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render only next button', () => {
    const component = mount(
      <Navigation
        current={0}
        total={12}
        onClickPrev={jest.fn()}
        onClickNext={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
