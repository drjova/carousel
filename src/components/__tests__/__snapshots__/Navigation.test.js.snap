// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render only next button 1`] = `
<div
  id="navigation"
>
  <button
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;

exports[`Navigation component should render only previous button 1`] = `
<div
  id="navigation"
>
  <button
    onClick={[MockFunction]}
  >
    Prev
  </button>
</div>
`;

exports[`Navigation component should render with both buttons 1`] = `
<div
  id="navigation"
>
  <button
    onClick={[MockFunction]}
  >
    Prev
  </button>
  <button
    onClick={[MockFunction]}
  >
    Next
  </button>
</div>
`;
